nexus/src/external_api/console_api.rs

```diff
@@ -601,7 +601,13 @@ pub async fn session_me(
         // not clear what the advantage would be.
         let opctx = OpContext::for_external_api(&rqctx).await?;
         let user = nexus.silo_user_fetch_self(&opctx).await?;
-        Ok(HttpResponseOk(user.into()))
+        let groups = nexus.silo_user_fetch_groups(&opctx).await?;
+        Ok(HttpResponseOk(views::SessionMe {
+            id: user.id(),
+            display_name: user.external_id,
+            silo_id: user.silo_id,
+            group_ids: groups.iter().map(|g| g.silo_group_id).collect(),
+        }))
```

Comment:
I think `views::SessionMe::new(user, groups)` would make sense here, but I struggled to decide where to define that. The obviously place would be right next to the definition of `SessionMe` in `views.rs`, but that would be the first time anything in that file refers to DB models. Could be fine, or maybe it offends our sense of propriety. Generally we seem to prefer doing our `impl From` for these views next to the DB model, which to be honest is not what I prefer. I like to think of the view as "knowing about" the model but the model not knowing about the view.

Reply:
In fact, there does not seem to be a way to import `db` into `nexus/types/src/external_api/views.rs`. So it kind of has to be defined here in the app, but I can't `impl` a constructor for it because it comes from another crate.

Reply:
this seems to point to a  `fn Nexus::session_me(..) -> views::SessionMe`, no?

Reply:
I considered that too. The problem is the app layer things methods all return DB models, not views, and it's up to the `From` impls to turn those into views in the entrypoints. This is a pretty rare case where we're making a view out of two models. I don't think we do that very often. So if it can't go in `nexus/src/app`, it would basically be a helper that sits right next to the route handler right here, at which point I figure we might as well inline it (leave as is) since it's not going to be used anywhere else.